File: agent/verticals/education.py
"""Education vertical definition."""

from agent.verticals.base import VerticalDefinition, tabs

VERTICAL = VerticalDefinition(
    key="education",
    label="Education",
    risk_level="low",
    entity_label_singular="course",
    entity_label_plural="courses",
    default_plan_label="Education plan",
    crm_tabs=tabs(
        ("overview", "Overview"),
        ("readiness", "Readiness"),
        ("catalog", "Courses"),
        ("leads", "Leads"),
        ("activity", "Conversations"),
        ("prompt", "Prompt"),
        ("controls", "Controls"),
    ),
    entity_types=("course", "program", "certificate", "instructor", "learning_path", "cohort"),
    readiness_checks=("courses", "syllabus", "pricing", "enrollment", "lead_capture", "policies"),
    action_types=(
        "SHOW_ENTITIES",
        "SORT_ENTITIES",
        "BUILD_LEARNING_PATH",
        "CHECK_PREREQUISITES",
        "START_ENROLLMENT",
        "REQUEST_COUNSELOR_CALLBACK",
        "CAPTURE_LEAD",
    ),
)

File: crm/src/verticals/definitions/education.ts
import { tab } from '../shared';
import type { CrmVerticalDefinition } from '../types';

export const educationVertical: CrmVerticalDefinition = {
  key: 'education',
  label: 'Education',
  riskLevel: 'low',
  entityLabelSingular: 'course',
  entityLabelPlural: 'courses',
  defaultPlanLabel: 'Education plan',
  clientTabs: [
    tab('overview', 'Overview'),
    tab('readiness', 'Readiness'),
    tab('catalog', 'Courses'),
    tab('leads', 'Leads'),
    tab('activity', 'Conversations'),
    tab('prompt', 'Prompt'),
    tab('controls', 'Controls'),
  ],
  entityTypes: ['course', 'program', 'certificate', 'instructor', 'learning_path', 'cohort'],
  readinessChecks: ['courses', 'syllabus', 'pricing', 'enrollment', 'lead_capture', 'policies'],
};
